### DISCUSSION TITLE
How can I put a wildcard inside a cosmetic filter?

### DISCUSSION THREAD

**Discussion by punisher018** [uBlockOrigin/uAssets#22713] [Q&A] [answered]
For example, imdb.com occasionally has a sign in pop up. I block this with UbO's element picker. I can't reach a definite solution because the code in the middle is different each time, like the filters below. I looked [here](https://www.reddit.com/r/uBlockOrigin/wiki/index/#wiki_with_.22wildcards.22_.28attribute_selectors.29) but I couldn't find exactly what I wanted. 

```
www.imdb.com##.ipc-list-card--base.cHimVI.sc-ArjuK.ipc-coachmark.ipc-list-card
www.imdb.com##.ipc-list-card--base.biLDlp.sc-laZMRZ.ipc-coachmark.ipc-list-card
www.imdb.com##.ipc-list-card--base.exJutK.sc-jgPygj.ipc-coachmark.ipc-list-card
```

Can the regex solution used for domain addresses be used for the cosmetic part?

**Comment by gorhill** [comment, marked as answer]
Just remove the whole class names which have random parts?

    www.imdb.com##.ipc-list-card--base.ipc-coachmark.ipc-list-card

**Reply by punisher018** [reply]
I didn't know that. It's actually quite simple. Thanks 😄

**Reply by garry-ut99** [reply]
> It's actually quite simple.

Simple and works on most sites, until you encounter a site with: `##.X4HhT.DuMIQc.AjY5Oe.nCP5yc.VfPpkd-LgbsSe-OWXEXe-dgl2Hf.VfPpkd-LgbsSe-OWXEXe-k8QpJ.VfPpkd-LgbsSe`, in such case removing random names will leave you with broken filter `##` , but even then, you usually still don't need a regex for it, instead you need to learn how to create good filters without using random names.

<hr>

> How can I put a wildcard inside a cosmetic filter?

Your question is wrong, because in your case a wildcard is not required, because it's better to just drop the random parts, but for the sake of answering the question, yes - it's possible, combine starts with + contains + ends with:
`www.imdb.com##[class^="ipc-list-card ipc-coachmark "][class*="sc-"][class$=" ipc-list-card--base"]` 
but it's better to combine the "contains" part into "starts with" part:
`www.imdb.com##[class^="ipc-list-card ipc-coachmark sc-"][class$=" ipc-list-card--base"]`

> Can the regex solution used for domain addresses be used for the cosmetic part?

Again, not required in your case, but for the sake of answering the question, yes - it's possible : https://github.com/uBlockOrigin/uBlock-issues/wiki/Procedural-cosmetic-filters#subjectmatches-attrarg but it's necessary only in rare and hard cases, and it's only for advanced users, you don't need it and you shouldn't use it until you become an advanced filter creator.

**Reply by punisher018** [reply]
Thank you very much. Actually, my first thought was whether to combine the beginning and end parts or the contain part as you wrote. Because if I specified only the beginning or the end part in the filter, there would be other blocked parts. It's very informative.
